internal/beater/beater.go

```diff
@@ -874,6 +936,13 @@ func (s *serverRunner) newFinalBatchProcessor(
 	return indexer, indexer.Close, nil
 }
 
+func eventBufferSize(memLimit float64) int {
+	if memLimit > 1 {
+		return int(256 * memLimit)
```

Comment:
Memory usage is till pretty low, we could likely increase from `256` to `512` or even further.